src/OpenTelemetry/Trace/Configuration/TracerBuilder.cs

```diff
@@ -50,6 +53,29 @@ public TracerBuilder SetSampler(ISampler sampler)
             return this;
         }
 
+        /// <summary>
+        /// Sets the <see cref="Resource"/> describing the app associated with all traces. Overwrites currently set resource. 
```

Comment:
TracerBuilder is single-use thing. It is not intended for any update scenarios. Basically you use it to construct factory and throw it away.

Runtime changes are not possible at all. So Set always overwrites and Merge is overkill :)

Reply:
I should have explained better. `Set` and `Merge` are different in semantics and needed. THe idea is that there are multiple resource collectors:
- One that reads environment variables
- one that set by user explicitly
- one that understands specific environment (ex Azure Web Apps)
etc.

One that will be pre-installed is that reads environment variables. User may choose to omit it if using `Set`.

Reply:
Something can merge all resources into one and set a default one on the TracerBuilder. Why does TracerBuilder needs to provide this capabilities?

Reply:
- I want to initialize TraceBuilder with the default one. And have an ability to override it. 
- Also I'd love to make calculation of resource lazy.
- and finally I can imagine some helper methods like AddAzureSupport() and AddK8SSupport() in TracerBuilder that adds whole bunch of resource and other things.

Thus I need some sort of resopurces pipeline builder. Alternative for this is to create a resources pipeline builder as an explicit type and set it on TracerBuilder. Are you advocating for this?

Reply:
Service name should be part of default resource, isn't it?
You should not reuse the same builder twice, so you would anyway construct one resource per builder.

AddAzureSupport or AddK8Ssupport are future and we are not designing them now. So how they will work - let decide later. Now it seems the scenario you want to have is to assign resource to the factory. So why can't we have one method for it and introduce complications only when they are needed.

Reply:
I like having both Set and Merge methods. It's clear they have different functions and it's nice to be able to replace the whole resource, or merge entries.